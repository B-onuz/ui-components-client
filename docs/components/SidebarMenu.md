# SidebarMenu (Component)

Barra lateral com menu de navegaÃ§Ã£o usado em conjunto com o componente `BaseLayout`

## Como utilizar

```js
import SidebarMenu from 'ui-components/lib/components/SidebarMenu'
```

```jsx
import React from 'react'
import SidebarMenu from 'ui-components/lib/components/SidebarMenu'
import SidebarMenuItem from 'ui-components/lib/components/SidebarMenuItem'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faHome, faDice, faCodeBranch, faAdjust, faKey } from '@fortawesome/free-solid-svg-icons'

const menuItems = [
  { icon: faHome, text: 'Home' },
  { icon: faDice, text: 'Dados' },
  { icon: faCodeBranch, text: 'Branch' },
  { icon: faAdjust, text: 'Adjusts Darkmode' },
  { icon: faKey, text: 'Chaves de acesso' },
]

export const SidebarMenuExample = () => (
  <SidebarMenu open={true}>
    {menuItems.map((item) => {
      return <SidebarMenuItem component={'header'} icon={<FontAwesomeIcon icon={item.icon} />} children={item.text} />
    })}
  </SidebarMenu>
)
```
