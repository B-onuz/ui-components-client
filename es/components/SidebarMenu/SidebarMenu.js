"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireWildcard(require("react"));

var _reactDom = _interopRequireDefault(require("react-dom"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _styledSystem = require("styled-system");

var _reflexbox = require("reflexbox");

var _MenuList = _interopRequireDefault(require("./MenuList"));

var _Button = _interopRequireDefault(require("../Button"));

var _this = void 0,
    _jsxFileName = "/Users/master/projects/bonuz/ui-components/src/components/SidebarMenu/SidebarMenu.js";

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject5() {
  var data = _taggedTemplateLiteralLoose(["\n  color: #ffffff;\n  text-decoration: none;\n  text-align: center;\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteralLoose(["\n  position: absolute;\n  bottom: 30px;\n  width: 100%;\n  display: flex;\n  flex-direction: column;\n  justify-content: center;\n  @media only screen and (max-height: 800px) {\n    position: unset;\n    padding-bottom: 4vh;\n  }\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteralLoose(["\n  ", "\n  position: absolute;\n  left: 0;\n  top: 0;\n  bottom: 0;\n  height: 100%;\n  background: #550e33 linear-gradient(89deg, #550e33 0%, #ff4f5d 100%) 0% 0% no-repeat padding-box;\n  transition: 0.2s linear;\n  z-index: 1002;\n  overflow: auto;\n  ", "\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteralLoose(["\n  height: 100%;\n  z-index: 9999;\n  ", "\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  visibility: hidden;\n  background: rgba(0, 0, 0, 0);\n  transition: 0.2s;\n  cursor: pointer;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

var Backdrop = _styledComponents["default"].div(_templateObject());

var styleOnlyMobile = function styleOnlyMobile(onlyMobile, styleCSS) {
  return onlyMobile ? styleCSS : "\n    @media only screen and (max-width: 1024px) {\n      " + styleCSS + "\n    }\n  ";
};

var WrapperSidebarMenu = _styledComponents["default"].nav(_templateObject2(), function (_ref) {
  var open = _ref.open,
      onlyMobile = _ref.onlyMobile;
  return !!open && styleOnlyMobile(onlyMobile, "\n      position: fixed;\n      top: 0;\n      right: 0;\n      bottom: 0;\n      left: 0;\n\n      " + Backdrop + " {\n        visibility: visible;\n        background: rgba(0, 0, 0, 0.3);\n        position: fixed;\n        top: 0;\n        right: 0;\n        bottom: 0;\n        left: 0;\n        z-index: 1001;\n      }\n  ");
});

var StyledSidebarMenu = (0, _styledComponents["default"])(_reflexbox.Box)(_templateObject3(), _styledSystem.layout, function (_ref2) {
  var open = _ref2.open;
  return open ? "\n    transform: translateX(0);\n  " : "\n    transform: translateX(-100%);\n  ";
});
var ContactButton = (0, _styledComponents["default"])(_reflexbox.Box)(_templateObject4());

var LogoutLink = _styledComponents["default"].a(_templateObject5());

var useRootSidebar = function useRootSidebar() {
  var _useState = (0, _react.useState)(),
      $rootModal = _useState[0],
      setRootModal = _useState[1];

  (0, _react.useEffect)(function () {
    var rootModal = document.getElementById('root-sidebar');

    if (rootModal) {
      setRootModal(rootModal);
    } else {
      var rootEl = document.createElement('div');
      rootEl.id = 'root-sidebar';
      setRootModal(rootEl);
      document.body.appendChild(rootEl);
    }
  });
  return {
    $rootModal: $rootModal
  };
};

var SidebarMenuRoot = function SidebarMenuRoot(_ref3) {
  var children = _ref3.children,
      _ref3$open = _ref3.open,
      open = _ref3$open === void 0 ? false : _ref3$open,
      _ref3$onlyMobile = _ref3.onlyMobile,
      onlyMobile = _ref3$onlyMobile === void 0 ? false : _ref3$onlyMobile,
      onClose = _ref3.onClose,
      _ref3$logo = _ref3.logo,
      logo = _ref3$logo === void 0 ? 'Logo' : _ref3$logo,
      _ref3$items = _ref3.items,
      items = _ref3$items === void 0 ? [] : _ref3$items,
      contactButton = _ref3.contactButton,
      logoutHref = _ref3.logoutHref,
      rest = _objectWithoutPropertiesLoose(_ref3, ["children", "open", "onlyMobile", "onClose", "logo", "items", "contactButton", "logoutHref"]);

  return /*#__PURE__*/_react["default"].createElement(WrapperSidebarMenu, {
    onlyMobile: onlyMobile,
    open: !!open,
    role: "menubar",
    "aria-label": "menu",
    "aria-expanded": !!open,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 112,
      columnNumber: 3
    }
  }, /*#__PURE__*/_react["default"].createElement(StyledSidebarMenu, _extends({
    role: "none",
    open: !!open
  }, rest, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 113,
      columnNumber: 5
    }
  }), logo, /*#__PURE__*/_react["default"].createElement(_MenuList["default"], {
    role: "menu",
    isOpen: !!open,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 115,
      columnNumber: 7
    }
  }, children), /*#__PURE__*/_react["default"].createElement(ContactButton, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 118,
      columnNumber: 7
    }
  }, contactButton, !!logoutHref && /*#__PURE__*/_react["default"].createElement(LogoutLink, {
    href: logoutHref,
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 120,
      columnNumber: 26
    }
  }, "Sair"))), /*#__PURE__*/_react["default"].createElement(Backdrop, {
    onClick: onClose,
    role: "button",
    "aria-label": "Fechar menu",
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 123,
      columnNumber: 5
    }
  }));
};

var SidebarMenuMobile = function SidebarMenuMobile(props) {
  var _useRootSidebar = useRootSidebar(),
      $rootModal = _useRootSidebar.$rootModal;

  if (!$rootModal) return null;
  return _reactDom["default"].createPortal( /*#__PURE__*/_react["default"].createElement(SidebarMenuRoot, _extends({}, props, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 131,
      columnNumber: 32
    }
  })), $rootModal);
};

var SidebarMenu = function SidebarMenu(props) {
  var SidebarComponent = props.onlyMobile ? SidebarMenuMobile : SidebarMenuRoot;
  return /*#__PURE__*/_react["default"].createElement(SidebarComponent, _extends({}, props, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 137,
      columnNumber: 10
    }
  }));
};

SidebarMenu.defaultProps = {
  width: '248px'
};
var _default = SidebarMenu;
exports["default"] = _default;