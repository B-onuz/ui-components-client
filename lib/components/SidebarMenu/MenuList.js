"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _styledComponents = _interopRequireDefault(require("styled-components"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  list-style: none;\n  padding: 0;\n  transition: 0.2s linear;\n  transform: translateX(100%);\n  opacity: 0.5;\n  ", "\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

var MenuList = _styledComponents["default"].ul(_templateObject(), function (_ref) {
  var isOpen = _ref.isOpen;
  return isOpen && "\n        transform: translateX(0);\n        opacity: 1;\n    ";
});

var _default = MenuList;
exports["default"] = _default;
module.exports = exports.default;