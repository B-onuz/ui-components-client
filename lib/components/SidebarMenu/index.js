"use strict";

exports.__esModule = true;
exports["default"] = void 0;

var _SidebarMenu = _interopRequireDefault(require("./SidebarMenu"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

var _default = _SidebarMenu["default"];
exports["default"] = _default;
module.exports = exports.default;