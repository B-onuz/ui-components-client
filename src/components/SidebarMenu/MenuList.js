import styled from 'styled-components'

const MenuList = styled.ul`
  list-style: none;
  padding: 0;
  transition: 0.2s linear;
  transform: translateX(100%);
  opacity: 0.5;
  ${({ isOpen }) =>
    isOpen &&
    `
        transform: translateX(0);
        opacity: 1;
    `}
`

export default MenuList
