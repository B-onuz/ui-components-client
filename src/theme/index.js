const colors = {
  red: '#db2828',
  lightRed: '#ef4d5a',
  mediumRed: '#FF4F5D',
  darkRed: '#9c2543',
  fullDarkRed: '#550e33',
  orange: '#f2711c',
  yellow: '#fbbd08',
  olive: '#b5cc18',
  green: '#21BA45',
  teal: '#00b5ad',
  blue: '#2185d0',
  facebookBlue: '#3B5998',
  blueLink: '#4285F4',
  violet: '#6435c9',
  purple: '#a333c8',
  pink: '#e03997',
  brown: '#a5673f',
  grey: '#767676',
  textGrey: '#8D8D8D',
  lightGrey: '#6D6D6D',
  mediumGrey: '#838383',
  darkGrey: '#9A9A9A',
  boldGray: '#55555A',
  black: '#1b1c1d',
  white: '#fdfdfd',
  warning: '#7A4D05',
  success: '#1E561F',
  info: '#0E566C',
  danger: '#973937',
  cyan: '#1CB5AD',
  disabledGrey: '#ffffff1f',
  disabledGreyText: '#ffffff4d',
}

colors.primary = colors.lightRed
colors.secondary = colors.fullDarkRed
colors.default = '#e0e1e2'

const breakpoints = ['40em', '52em', '64em', '80em']

breakpoints.sm = breakpoints[0]
breakpoints.md = breakpoints[1]
breakpoints.lg = breakpoints[2]
breakpoints.xl = breakpoints[3]

const fontSizes = [12, 14, 16, 20, 24, 32]

export default {
  fontFamily: `'Lato', sans-serif`,
  colors,
  fontSizes,
  breakpoints,
}
