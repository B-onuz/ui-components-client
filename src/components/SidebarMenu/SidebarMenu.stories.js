import React from 'react'
import SidebarMenu from './SidebarMenu'
import SidebarMenuItem from '../SidebarMenuItem'
import Button from '../Button'
import shortid from 'shortid'
import { Box, Flex } from 'reflexbox'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faHome, faDice, faCodeBranch, faAdjust, faKey, faPhone } from '@fortawesome/free-solid-svg-icons'

export default {
  title: 'SidebarMenu',
  component: SidebarMenu,
}

const menuItems = [
  { icon: faHome, text: 'Home' },
  { icon: faDice, text: 'Dados' },
  { icon: faCodeBranch, text: 'Branch' },
  { icon: faAdjust, text: 'Adjusts Darkmode' },
  { icon: faKey, text: 'Chaves de acesso' },
  { icon: faHome, text: 'Home' },
  { icon: faDice, text: 'Dados' },
  { icon: faCodeBranch, text: 'Branch' },
  { icon: faAdjust, text: 'Adjusts Darkmode' },
  { icon: faKey, text: 'Chaves de acesso' },
]

export const SidebarMenuExample = () => (
  <SidebarMenu
    open={true}
    contactButton={
      <Button style={{ backgroundColor: 'transparent' }} color="white" outline m={3}>
        <Flex alignItems="center">
          <Box mr={2}>
            <FontAwesomeIcon icon={faPhone} color="#4AB77C" size="lg" />
          </Box>
          <Flex width={1} justifyContent="center">
            +11 22 33333-3333
          </Flex>
        </Flex>
      </Button>
    }
    logoutHref={'google.com'}
  >
    {menuItems.map((item) => {
      return (
        <SidebarMenuItem
          key={`${item.text}-${shortid.generate()}`}
          component={'header'}
          icon={<FontAwesomeIcon icon={item.icon} />}
          children={item.text}
        />
      )
    })}
  </SidebarMenu>
)

export const SidebarMenuOnlyMobile = () => (
  <SidebarMenu
    open={true}
    onlyMobile
    contactButton={
      <Button style={{ backgroundColor: 'transparent' }} color="white" outline m={3}>
        <Flex alignItems="center">
          <Box mr={2}>
            <FontAwesomeIcon icon={faPhone} color="#4AB77C" size="lg" />
          </Box>
          <Flex width={1} justifyContent="center">
            +11 22 33333-3333
          </Flex>
        </Flex>
      </Button>
    }
  >
    {menuItems.map((item) => {
      return (
        <SidebarMenuItem
          key={`${item.text}-${shortid.generate()}`}
          component={'header'}
          icon={<FontAwesomeIcon icon={item.icon} />}
          children={item.text}
        />
      )
    })}
  </SidebarMenu>
)
