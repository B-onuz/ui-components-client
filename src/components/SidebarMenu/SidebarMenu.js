import React, { useState, useEffect } from 'react'
import ReactDOM from 'react-dom'
import styled from 'styled-components'
import { layout } from 'styled-system'
import { Box } from 'reflexbox'
import MenuList from './MenuList'
import Button from '../Button'

const Backdrop = styled.div`
  visibility: hidden;
  background: rgba(0, 0, 0, 0);
  transition: 0.2s;
  cursor: pointer;
`

const styleOnlyMobile = (onlyMobile, styleCSS) =>
  onlyMobile
    ? styleCSS
    : `
    @media only screen and (max-width: 1024px) {
      ${styleCSS}
    }
  `

const WrapperSidebarMenu = styled.nav`
  height: 100%;
  z-index: 9999;
  ${({ open, onlyMobile }) =>
    !!open &&
    styleOnlyMobile(
      onlyMobile,
      `
      position: fixed;
      top: 0;
      right: 0;
      bottom: 0;
      left: 0;

      ${Backdrop} {
        visibility: visible;
        background: rgba(0, 0, 0, 0.3);
        position: fixed;
        top: 0;
        right: 0;
        bottom: 0;
        left: 0;
        z-index: 1001;
      }
  `
    )}
`

const StyledSidebarMenu = styled(Box)`
  ${layout}
  position: absolute;
  left: 0;
  top: 0;
  bottom: 0;
  height: 100%;
  background: #550e33 linear-gradient(89deg, #550e33 0%, #ff4f5d 100%) 0% 0% no-repeat padding-box;
  transition: 0.2s linear;
  z-index: 1002;
  overflow: auto;
  ${({ open }) =>
    open
      ? `
    transform: translateX(0);
  `
      : `
    transform: translateX(-100%);
  `}
`

const ContactButton = styled(Box)`
  position: absolute;
  bottom: 30px;
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  @media only screen and (max-height: 800px) {
    position: unset;
    padding-bottom: 4vh;
  }
`

const LogoutLink = styled.a`
  color: #ffffff;
  text-decoration: none;
  text-align: center;
`

const useRootSidebar = () => {
  const [$rootModal, setRootModal] = useState()

  useEffect(() => {
    const rootModal = document.getElementById('root-sidebar')
    if (rootModal) {
      setRootModal(rootModal)
    } else {
      const rootEl = document.createElement('div')
      rootEl.id = 'root-sidebar'
      setRootModal(rootEl)
      document.body.appendChild(rootEl)
    }
  })

  return { $rootModal }
}

const SidebarMenuRoot = ({ children, open = false, onlyMobile = false, onClose, logo = 'Logo', items = [], contactButton, logoutHref, ...rest }) => (
  <WrapperSidebarMenu onlyMobile={onlyMobile} open={!!open} role="menubar" aria-label="menu" aria-expanded={!!open}>
    <StyledSidebarMenu role="none" open={!!open} {...rest}>
      {logo}
      <MenuList role="menu" isOpen={!!open}>
        {children}
      </MenuList>
      <ContactButton>
        {contactButton}
        {!!logoutHref && <LogoutLink href={logoutHref}>Sair</LogoutLink>}
      </ContactButton>
    </StyledSidebarMenu>
    <Backdrop onClick={onClose} role="button" aria-label="Fechar menu" />
  </WrapperSidebarMenu>
)

const SidebarMenuMobile = (props) => {
  const { $rootModal } = useRootSidebar()
  if (!$rootModal) return null

  return ReactDOM.createPortal(<SidebarMenuRoot {...props} />, $rootModal)
}

const SidebarMenu = (props) => {
  const SidebarComponent = props.onlyMobile ? SidebarMenuMobile : SidebarMenuRoot

  return <SidebarComponent {...props} />
}

SidebarMenu.defaultProps = {
  width: '248px',
}

export default SidebarMenu
