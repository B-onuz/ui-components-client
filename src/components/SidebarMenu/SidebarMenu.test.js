import React from 'react'
import SidebarMenu from './SidebarMenu.js'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import theme from '../../theme'
import 'jest-styled-components'

test('SidebarMenu changes the class when hovered', () => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <SidebarMenu />
    </ThemeProvider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
