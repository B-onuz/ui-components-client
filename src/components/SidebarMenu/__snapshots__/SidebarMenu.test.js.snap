// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarMenu changes the class when hovered 1`] = `
.c2 {
  list-style: none;
  padding: 0;
  -webkit-transition: 0.2s linear;
  transition: 0.2s linear;
  -webkit-transform: translateX(100%);
  -ms-transform: translateX(100%);
  transform: translateX(100%);
  opacity: 0.5;
}

.c4 {
  visibility: hidden;
  background: rgba(0,0,0,0);
  -webkit-transition: 0.2s;
  transition: 0.2s;
  cursor: pointer;
}

.c0 {
  height: 100%;
  z-index: 9999;
}

.c1 {
  width: 248px;
  position: absolute;
  left: 0;
  top: 0;
  bottom: 0;
  height: 100%;
  background: #550e33 linear-gradient(89deg,#550e33 0%,#ff4f5d 100%) 0% 0% no-repeat padding-box;
  -webkit-transition: 0.2s linear;
  transition: 0.2s linear;
  z-index: 1002;
  overflow: auto;
  -webkit-transform: translateX(-100%);
  -ms-transform: translateX(-100%);
  transform: translateX(-100%);
}

.c3 {
  position: absolute;
  bottom: 30px;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media only screen and (max-height:800px) {
  .c3 {
    position: unset;
    padding-bottom: 4vh;
  }
}

<nav
  aria-expanded={false}
  aria-label="menu"
  className="c0"
  open={false}
  role="menubar"
>
  <div
    className="c1 css-13rhp43"
    open={false}
    role="none"
  >
    Logo
    <ul
      className="c2"
      role="menu"
    />
    <div
      className="c3 css-vurnku"
    />
  </div>
  <div
    aria-label="Fechar menu"
    className="c4"
    role="button"
  />
</nav>
`;
